crates/storage/db/src/implementation/mdbx/mod.rs

```diff
@@ -103,17 +103,31 @@ pub struct DatabaseArguments {
     /// MDBX allows up to 32767 readers (`MDBX_READERS_LIMIT`). This arg is to configure the max
     /// readers.
     max_readers: Option<u64>,
+    /// Defines the synchronization strategy used by the MDBX database when writing data to disk.
+    ///
+    /// This determines how aggressively MDBX ensures data durability versus prioritizing performance.
+    /// The available modes are:
+    ///
+    /// - [`SyncMode::Durable`]: Ensures all transactions are fully flushed to disk before they are considered committed.  
+    ///   This provides the highest level of durability and crash safety but may have a performance cost.
+    /// - [`SyncMode::SafeNoSync`]: Skips certain fsync operations to improve write performance.  
+    ///   This mode still maintains database integrity but may lose the most recent transactions
+    ///   if the system crashes unexpectedly.
+    ///
+    /// Choose `Durable` if consistency and crash safety are critical (e.g., production environments).  
+    /// Choose `SafeNoSync` if performance is more important and occasional data loss is acceptable (e.g., testing or ephemeral data).
+    sync_mode: SyncMode,
 }
 
 impl Default for DatabaseArguments {
     fn default() -> Self {
-        Self::new(ClientVersion::default())
+        Self::new(ClientVersion::default(), SyncMode::Durable)
     }
 }
 
 impl DatabaseArguments {
     /// Create new database arguments with given client version.
-    pub fn new(client_version: ClientVersion) -> Self {
+    pub fn new(client_version: ClientVersion, sync_mode: SyncMode) -> Self {
```

Comment:
instead of adding this to new, can we init this always with durable

and add another `fn with_sync_mode` that sets this